### PR TITLE
Bugfix: Dynamic Email Content filters using Custom Fields

### PR DESCRIPTION
**Pull request by dongilbert**
[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | Y
| New feature? | N
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | #3719
| BC breaks? | N/A
| Deprecations? | N/A

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

This PR fixes dynamic content filters that use custom fields as their filters. It also fixes an issue where segment filters and DEC filters used the label for multiselects rather than the values.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Create a Multi-select custom field with multiple values, open a contact, and select a value for the custom field you created. Save the contact.
2. Create an email with a DEC item, with a variant using the created multi-select field as the filter.
3. Send that email to the contact you modified. You'll see that the default content is returned, instead of the matching variant content.

#### Steps to test this PR:
1. Apply PR
2. Retest. The received email will have the appropriate variant.

### CODE REVIEW

File: app/bundles/CoreBundle/Assets/js/libraries/4j.codemirror.js [hunk 1]
```diff
@@ -3586,7 +3586,11 @@
     }
     if (clickInGutter(cm, e)) return;
     var start = posFromMouse(cm, e);
-    window.focus();
+
+    if (typeof Mautic != 'undefined' && Mautic.codeMode != true) {
+      console.log(Mautic);
```
Comment (escopecz) [inline app/bundles/CoreBundle/Assets/js/libraries/4j.codemirror.js]: This is probably a forgotten debug code.

Comment (dongilbert) [inline app/bundles/CoreBundle/Assets/js/libraries/4j.codemirror.js]: Need to update this as the JS fixes for dwc are not longer required - fixed in another PR by @virlatinus

**Comment by virlatinus** [comment]
I checked it out to my current staging and I'm getting an error
`mautic.CRITICAL: Uncaught PHP Exception Symfony\Component\Debug\Exception\ContextErrorException: "PHP Warning - Invalid argument supplied for foreach()" at /var/www/mautic-fork/app/bundles/CoreBundle/Helper/AbstractFormFieldHelper.php`

**Comment by virlatinus** [comment]
I followed the steps again to test: created the custom field and when editing a contact I get the error
I then deleted the custom field and the error is gone.

**Comment by virlatinus** [comment]
This code is already on #3743

**Pull request closed without merging**